### PR TITLE
Added ticket proof validation to client.

### PR DESCRIPTION
**Pull request by micwallace**
Hey @nicktaras 

This PR adds ZK proof validation into the TN client authenticate function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,15 +26,15 @@
   },
   "homepage": "https://github.com/TokenScript/token-negotiator#readme",
   "dependencies": {
-    "@tokenscript/attestation": "^1.0.7-alpha",
+    "@tokenscript/attestation": "^0.3.1",
```
Comment (samwarwick) [inline package.json:29]: The versioning scheme has been puzzling me. I noticed earlier today at https://www.npmjs.com/package/@tokenscript/attestation that the numbering had seemingly gone backwards. Is this intentional/by-design?

Comment (nicktaras) [reply package.json:29]: Hi @samwarwick, GP. Let's keep to one approach.

It seems in this case the method of versioning has changed (from alpha .... which would then start at 1.0.0 once production ready / to 0.3.1 which will work towards 1.0.0). There are semver tools that can prompt us of the type of changes are being added e.g. minor, major / breaking changes, bug fix. There's an issue inside Git to implement this before we get to being prod ready.

Comment (samwarwick) [reply package.json:29]: Adding the existing Token Negotiator ticket for reference: https://github.com/TokenScript/token-negotiator/issues/65. Might need something similar for Attestation.

Comment (micwallace) [reply package.json:29]: > The versioning scheme has been puzzling me. I noticed earlier today at https://www.npmjs.com/package/@tokenscript/attestation that the numbering had seemingly gone backwards. Is this intentional/by-design?

Hi @samwarwick this was done to keep the major/minor version in sync with the Java attestation library.

**Comment by micwallace** [comment]
@jot2re You may want to have a look at this too: 

https://github.com/TokenScript/token-negotiator/pull/160/files#diff-24f1ac19d22913269d99a8a5afa4c13aa7fd2c8b9858529046424841ca6421daR403

Uses code from attestation branch here:
https://github.com/TokenScript/attestation/blob/e828eca3dcf36bf57fcbf95a93310b94b06f865a/src/main/javascript/crypto/src/Authenticator.ts#L130

**Comment by jot2re** [comment]
Once the feedback on[ Attestation PR 239](https://github.com/TokenScript/attestation/pull/239 ) gets handled, then it looks fine to me

**Comment by nicktaras** [comment]
> 

@micwallace, please feel free to fix any src/client/index.ts conflicts and merge this when ready (Attestation PR is ready etc).

**Pull request merged**